packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -899,3 +893,17 @@ function computeRoutePriority(config: AstroConfig): RoutePriorityOverride {
 	}
 	return 'legacy';
 }
+
+function joinSegments(segments: RoutePart[][]): string {
+	const arr = segments.map(function d(segment): string {
+		const [{ dynamic, content }] = segment;
+
+		return segment.length > 1
+			? segment.map((s) => d([s])).join('')
```

Comment:
Could you confirm I have got this right?

This one line is the fix.

Previously, we ignored all but the first `RoutePart`.

Now, each `RoutePart` gets passed to this `d()` function. And the function ultimately still wraps the dynamic `RouteParts` in square brackets same as it was done before.

Reply:
Sure thing! You've got it right.
Would there be any issues with this?

Reply:
No, this is great - just took me a while to understand the recursive map, so I thought I'd double check.